src/components/funFacts/FunFactComponent.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { funFacts, FunFact } from "./funfacts"
+import { useState } from "react"
+import "./FunFactPage.scss"
+
+interface FunFactsProps {
+    funFacts: FunFact[];
+}
+
+export const FunFactComponent: React.FunctionComponent<FunFactsProps> = ({ funFacts }) => {
+
+  const [selectedFact, setSelectedFact] = useState<number>(0);
```

Comment:
Don't need to specify the type when it can be inferred from the initial value:
```suggestion
  const [selectedFact, setSelectedFact] = useState(0);
```

Reply:
Also I think the name should reflect the usage as an ID:
```suggestion
  const [selectedFactId, setSelectedFactId] = useState<number>(0);
```
or (as I advised Kostas and Bharathi to do) you can do away with IDs entirely and just use the index of the fact in the fun fact array!
```suggestion
  const [selectedFactIndex, setSelectedFactIndex] = useState<number>(0);
```

Reply:
Can I leave this, as we are using selectedFactId=0 to show none of the fact details